### PR TITLE
PLT-2722 Sanitize sensitive data of out config file for the system console

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

**Comment by coreyhulen** [comment]
+1

File: api/admin.go [hunk 1]
```diff
@@ -160,8 +162,10 @@ func saveConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	utils.SaveConfig(utils.CfgFileName, cfg)
 	utils.LoadConfig(utils.CfgFileName)
-	json := utils.Cfg.ToJson()
-	w.Write([]byte(json))
+
+	rdata := map[string]string{}
```
Comment (crspeller) [inline api/admin.go]: Why are we writing this out?

File: model/config.go [hunk 2]
```diff
@@ -597,10 +599,38 @@ func (o *Config) IsValid() *AppError {
...
-	options["fullname"] = me.PrivacySettings.ShowFullName
-	options["email"] = me.PrivacySettings.ShowEmailAddress
+	options["fullname"] = o.PrivacySettings.ShowFullName
+	options["email"] = o.PrivacySettings.ShowEmailAddress
 
 	return options
 }
+
+func (o *Config) Sanitize() {
+	if len(*o.LdapSettings.BindPassword) > 0 {
```
Comment (crspeller) [inline model/config.go]: Why even check? Why not just set?

Comment (jwilander) [inline model/config.go]: We want the field to be blank in the System Console when no password has been set.

File: api/admin.go (hunk 1 above)
Comment (jwilander) [inline api/admin.go]: Need some sort of response, we do this in multiple places

Comment (crspeller) [inline api/admin.go]: Do we? I think I have removed these unnecessary responses in other places.

Comment (jwilander) [inline api/admin.go]: Scroll through `api/admin.go` they're all over. Let's discuss in the dev meeting about what to do about them. I don't think we should have any APIs that don't have some sort of response.

**Comment by crspeller** [comment]
+1

**Pull request merged**